src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -577,11 +581,12 @@ def listFileBlockLocation(self, fileBlockName, dbsOnly = False, phedexNodes=Fals
                 msg += "%s\n" % str(ex)
                 raise Exception(msg)
 
-            if not blockInfo or len(blockInfo) != len(blockNames): #if we couldnt get data location from PhEDEx, try to look into origin site location from dbs
-                dbsOnly = True
-                blockNames = set(blockNames) - set(blockInfo) #get the blocks we did not find information in phedex
+            if blockInfo and len(blockInfo) == len(blockNames):
+                phedexOnly = True
```

Comment:
From yesterday's CRAB3 meeting, I understood @ericvaandering was not in favor of changing the default behavior of the code, arguing that merge jobs may rely on it.